backend/tracim_backend/lib/utils/authorization.py

```diff
@@ -43,177 +47,266 @@ def permits(self, context, principals, permision):
+            .get_one_by_slug(content.type).slug
+        if current_content_type_slug in self.allowed_content_type_list:
+            return True
+        raise ContentTypeNotAllowed()
 
-def require_content_types(content_types_slug: typing.List[str]) -> typing.Callable:  # nopep8
+
+class CommentOwnerChecker(AuthorizationChecker):
     """
-    Restricts access to specific file type or raise an exception.
-    Check role for candidate_workspace.
-    :param content_types_slug: list of slug of content_types
-    :return: decorator
+    Check if current_user is owner of current_comment
     """
-    def decorator(func: typing.Callable) -> typing.Callable:
-        @functools.wraps(func)
-        def wrapper(self, context, request: 'TracimRequest') -> typing.Callable:
-            content = request.current_content
-            current_content_type_slug = content_type_list.get_one_by_slug(content.type).slug  # nopep8
-            if current_content_type_slug in content_types_slug:
-                return func(self, context, request)
-            raise ContentTypeNotAllowed()
-        return wrapper
-    return decorator
+    def __init__(self):
+        super().__init__()
+
+    def check(
+        self,
+        tracim_context: TracimContext
+    ) -> bool:
 
+        if tracim_context.current_comment.owner.user_id\
+                == tracim_context.current_user.user_id:
+            return True
+        raise UserIsNotContentOwner(
+            'user {} is not owner of comment  {}'.format(
+                tracim_context.current_user.user_id,
+                tracim_context.current_comment.content_id
+            )
+        )
 
-def require_comment_ownership_or_role(
-        minimal_required_role_for_owner: int,
-        minimal_required_role_for_anyone: int,
-) -> typing.Callable:
+
+class OrAuthorizationChecker(AuthorizationChecker):
     """
-    Decorator for view to restrict access of tracim request if role is
-    not high enough and user is not owner of the current_content
-    :param minimal_required_role_for_owner: minimal role for owner
-    of current_content to access to this view
-    :param minimal_required_role_for_anyone: minimal role for anyone to
-    access to this view.
-    :return:
+    Check multiple auth_checker with a logical operator "or"
+    return last exception found in list of checker
     """
+    def __init__(self, *authorization_checkers):
+        super().__init__()
+        self.authorization_checkers = authorization_checkers
+
+    def check(
+        self,
+        tracim_context: TracimContext
+    ) -> bool:
+        exception_to_raise = None
+        for authorization_checker in self.authorization_checkers:
+            try:
+                authorization_checker.check(
+                    tracim_context=tracim_context,
+                )
+                return True
+            except Exception as e:
+                exception_to_raise = e
+
+        raise exception_to_raise
```

Comment:
It seems to me complicated, we should be able to return http_code and probably special error code in body for those issues.

Reply:
I'm not sure get all error exception make also always sense. This means doing extra-check which are not required (we know auth as failed) and being really explicit about security mecanism ( it means saying to normal user " you can do that because you are not admin and you profile is not high enough for example", i'm not sure we should explain each time to user that admin have special access here.

Reply:
see #1226